lib/topological_inventory/ansible_tower/receptor/collector.rb

```diff
@@ -69,12 +74,18 @@ def wait_for_collected_data
 
         # Wait until all async responses are sent
         while entity_types_collected_cnt.value < service_catalog_entity_types.size
-          if last_save_at.nil? || (last_save_at < ReceptorController::Client::Configuration.default.response_timeout.minutes.ago && last_save_at >= max_wait_sync_threshold.minutes.ago)
+          if last_response_at.nil? && refresh_started_at >= max_wait_sync_threshold.minutes.ago
+            logger.warn("[ASYNC] Collector for source_uid: #{source}: No response received since #{refresh_started_at}")
+            sleep(60)
+          elsif last_response_at < ReceptorController::Client::Configuration.default.response_timeout.minutes.ago && last_response_at >= max_wait_sync_threshold.minutes.ago
```

Comment:
Will this condition throw an error if `last_response_at` is `nil`?  (Same for the last `elsif` block)

```ruby
nil < 100
=> NoMethodError (undefined method `<' for nil:NilClass)
```

Should there be a higher level condition here that checks for `if last_response_at.nil?` and then do further processing inside of that?

```ruby
while entity_types_collected_cnt.value < service_catalog_entity_types.size
  if last_response_at.nil?
    ...
  else
    ...
  end
end
```

Reply:
ou, you're right